lib/chef/provisioning/fog_driver/providers/softlayer.rb

```diff
@@ -28,6 +34,96 @@ def self.compute_options_for(provider, id, config)
 
             [result, id]
           end
+
+          def bootstrap_options_for(action_handler, machine_spec, machine_options)
+            opts = super
+            if opts[:bare_metal]
+              # these don't work with fog-softlayer bare_metal for some reason...
```

Comment:
Not clear what "don't work" means as it isn't stated, that said, it's worth noting that ordering a bare metal machine via the API doesn't immediately return with `server.id`, but only the `server.uid` property.  It can take up to an hour for the API to begin to return an `id` for the machine that matches that `uid`.